### PR TITLE
COHub summary and detail breakout to multi-sf

### PR DESCRIPTION
**Pull request by esc1144**
Added a separate SF for the COH recommendation calls. I was able to reuse the same Lambda function with some minor branching mods at the handler to determine if it was for the summary SF versus the detail. 
Reused the general SF role but needed to add object read permission in S3 for the list of recommendations to iterate over
Created Glue Crawlers for each recommendation action type since the schema can vary

An alternative approach is to consolidate into a single SF but it would mean this one module does not reuse the core SF template that every other module uses. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: data-collection/utils/lint.sh [hunk 1]
```diff
@@ -39,7 +39,7 @@ for file in $yaml_files; do
         echo -e "checkov      ${GREEN}OK${NC}"  | awk '{ print "\t" $0 }'
     fi
 
-    if [ "$(basename $file)" == "${exclude_files[0]}" ] || [ "$(basename $file)" == "${exclude_files[1]}" ] || [ "$(basename $file)" == "${exclude_files[2]}" ]; then
+    if [ "$(basename $file)" == "${exclude_files[0]}" ] || [ "$(basename $file)" == "${exclude_files[1]}" ] || [ "$(basename $file)" == "${exclude_files[2]}" ] || [ "$(basename $file)" == "${exclude_files[3]}" ]; then
```
Comment (iakov-aws) [inline data-collection/utils/lint.sh:42]:
```suggestion
    if [ "$(basename $file)" == "${exclude_files[0]}" ] || [ "$(basename $file)" == "${exclude_files[1]}" ] || [ "$(basename $file)" == "${exclude_files[2]}" ]; then
```

File: data-collection/deploy/module-cost-optimization-hub.yaml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 AWSTemplateFormatVersion: "2010-09-09"
 Description: Retrieves AWS Cost Optimization Hub Data across AWS organization
+Transform: 'AWS::LanguageExtensions'
```
Comment (iakov-aws) [inline data-collection/deploy/module-cost-optimization-hub.yaml:3]:
```suggestion
```

File: data-collection/deploy/module-cost-optimization-hub.yaml [hunk 3]
```diff
@@ -115,57 +124,114 @@ Resources:
...
+          h.setFormatter(logging.Formatter('[%(levelname)s] %(message)s'))
+          logger.addHandler(h)
           logger.setLevel(getattr(logging, os.environ.get('LOG_LEVEL', 'INFO').upper(), logging.INFO))
 
           BUCKET_NAME = os.environ['BUCKET_NAME']
           ROLENAME = os.environ['ROLENAME']
           PREFIX = os.environ['PREFIX']
+          DETAIL_SM_ARN = os.environ['DETAIL_SM_ARN']
           TMP_FILE = "/tmp/data.json"
-          NEED_DETAILED = True # set it to False to fallback to collection of only summaries if any perf issue or timeout
```
Comment (iakov-aws) [inline data-collection/deploy/module-cost-optimization-hub.yaml:126]: can we still keep option not to pull details?

Comment (esc1144) [reply data-collection/deploy/module-cost-optimization-hub.yaml:126]: Absolutely. Good idea for some use cases. I added it as configurable via the os param

**Review by iakov-aws — APPROVED** [review]

**Review by iakov-aws — APPROVED** [review]

**Pull request merged**